curvefs/src/client/kvclient/kvclient_manager.h

```diff
@@ -98,12 +95,17 @@ class KVClientManager {
 
     void Get(std::shared_ptr<GetKVCacheTask> task);
 
+    std::shared_ptr<KVClientMetric> &GetClientMetric() {
+        return kvClientMetric_;
+    }
+
```

Comment:
> Return references are invalid in c++, clang and gcc will use `RVO`.

IMHO, this isn't about RVO, the reference here means if you write code like this
```cpp
auto& metric = xxx.GetClienMetric();
```
then, internal `kvClientMetric_` doesn't get copied.

Otherwise, if return by value, then `auto& metric = xxx.GetClienMetric()` will be ill-formed, because non-const lvalue can't bind to rvalue, so that, we have to copy `kvClientMetric_` once which seems unnecessary.

Reply:
>>  Otherwise, if return by value, then `auto& metric = xxx.GetClienMetric()` will be ill-formed, because non-const lvalue can't bind to rvalue, so that, we have to copy `kvClientMetric_` once which seems unnecessary.

Of course, avoid coping is good. but sometimes if you avoid copying, seems danger in multi thread. Because you don't copy, the object you are manipulating is likely to be destructured